vyper/codegen/core.py

```diff
@@ -124,33 +114,22 @@ def make_byte_array_copier(destination, source, pos=None):
         return builder.resolve(copy_bytes(destination, src, n_bytes, max_bytes, pos=pos))
 
 
-def _word_size(location):
+def _wordsize(location):
     if location in ("memory", "calldata", "code"):
         return 32
     if location == "storage":
         return 1
     raise CompilerPanic(f"invalid location {location}")  # pragma: no test
 
 
-# Copy dynamic array word-for-word (including layout)
-# TODO this code is very similar to make_byte_array_copier,
-# try to refactor.
-def make_dyn_array_copier(dst, src, context, pos=None):
+def _dynarray_make_setter(dst, src, context, pos=None):
     assert isinstance(src.typ, DArrayType)
     assert isinstance(dst.typ, DArrayType)
 
-    if src.typ.count > dst.typ.count:
-        raise TypeMismatch(f"Cannot cast from {src.typ} to {dst.typ}")
-
-    # stricter check for zeroing
-    if src.value is None and src.typ.count != dst.typ.count:
-        raise CompilerPanic(f"Bad type for clearing bytes: expected {dst.typ} but got {src.typ}")
-
     with src.cache_when_complex("_src") as (b1, src):
-        if (
-            src.encoding in (Encoding.ABI, Encoding.JSON_ABI)
-            and src.typ.subtype.abi_type.is_dynamic()
-        ):
+        if src.typ.subtype.abi_type.is_dynamic():
+            # if the subtype is dynamic, for performance reasons
```

Comment:
For compiling performance? Or for optimal code performance? (cheaper gas, or smaller code?)

Reply:
better code. will clarify